### ISSUE TITLE
PDF generates multiple blank pages (in Chrome Canary)

### ISSUE THREAD

**Issue by calculuschild** [naturalcrit/homebrewery#2445]
In one of our recent bug reports to Chrome, an employee tipped me off that in an upcoming change to Chrome, they will be handling column breaks differently, and our site will end up generating lots of extra pages in PDFs as a result:

> The new fragmentation engine has much better support for multi-col/printing (better support for break-before:avoid for example).
> 
> While testing your site in Chrome Canary I noted that your testcase is producing additional pages. This is due to additional content producing additional columns (the "<p>&nbsp;</p>" and following content at the end of the page specifically). In our new fragmentation engine any "overflow" columns will be placed on subsequent pages which wasn't true previously.
> 
> A simple mitigation (which I think should work given how you structure your generator) is to add the additional rule:
> .page { contain: size }
> To you CSS.
> 
> This will tell our fragmentation engine that no (page) breaks are allowed within this content, and it'll place everything on one page.
> 
> This only works given you have explicitly set a fixed width/height for you pages.

I believe there is another similar comment from an older bug report where they also suggested we avoid duplicated use of `break:always`, etc. on our "column-break" element, since eventually it will start getting counted twice.

**Comment by calculuschild** [comment]
I'm not totally sure.

**Comment by Gazook89** [comment]
I just downloaded Canary and opened a brew.  I put filler text in that was maybe equivalent to 2 full pages with no page breaks (so just one page with overflow), and it went to the PDF printer with 59 blank white pages.  I reduced the filler text to just about 2 paragraphs over a full page, and got 9 pages of white.

I split the text into two pages (with `\page`) so the text fits neatly with no overflow-- the PDF print gave me those two pages but with a blank white page in between.

**Comment by calculuschild** [comment]
Ok so it's probably not flowing between divs as you would want, but rather triggering additional page breaks in the pdf.

**Comment by Gazook89** [comment]
Right--- I was doubtful that it could be as good as that, since i hadn't heard any squawking about it in web dev forums.

I added `contain: size;` to `.page` and it fixed the issue.  I haven't heard of the `contain` property before so I'm interested in finding out more about it, maybe it would be useful elsewhere in HB.

Should I submit a PR on this with the simple fix, or want to dig further into it yourself?  Or cross the bridge when we get there?

**Comment by calculuschild** [comment]
You can make a PR for this if you want. As long as it doesn't appear to break something else it seems like it's worth doing.

**Comment by G-Ambatte** [comment]
> You can make a PR for this if you want. As long as it doesn't appear to break something else it seems like it's worth doing.

I've created #2516 for this issue now.
I'm about to do dinner, but I'll submit a PR in a few hours unless someone else gets to it first.
